src/observer/index.js

```diff
@@ -182,18 +186,18 @@ function defineReactive (obj, key, val) {
         if (childOb) {
           childOb.dep.depend()
         }
-        if (_.isArray(val)) {
-          for (var e, i = 0, l = val.length; i < l; i++) {
-            e = val[i]
+        if (_.isArray(getter())) {
```

Comment:
Two things here:

1. The getter and setter should be invoked using `obj` as the `this` context, so `getter.call(obj)`.

2. The getter only needs to be called once inside the reactive getter. Calling it every time seems unnecessary.